hd/address.go

```diff
@@ -282,9 +282,20 @@ func CalcSha512(buf []byte) []byte {
 }
 
 func ReverseBytes(buf []byte) []byte {
-	res := []byte{}
-	for i := len(buf) - 1; i >= 0; i-- {
-		res = append(res, buf[i])
+	var res []byte
+	if len(buf) == 0 {
+		return res
+	}
+
+	// Walk till mid-way, swapping bytes from each end:
+	// b[i] and b[len-i-1]
```

Comment:
Good suggestion for simplicity. The downside is that it incurs a little more work than needs to be and the benchmarks reflect a 4.97% slow down
```diff
diff --git a/hd/address.go b/hd/address.go
index d7553a4..023b68b 100644
--- a/hd/address.go
+++ b/hd/address.go
@@ -287,15 +287,11 @@ func ReverseBytes(buf []byte) []byte {
 		return res
 	}
 
-	// Walk till mid-way, swapping bytes from each end:
-	// b[i] and b[len-i-1]
 	blen := len(buf)
 	res = make([]byte, blen)
-	mid := blen / 2
-	for left := 0; left <= mid; left++ {
-		right := blen - left - 1
-		res[left] = buf[right]
-		res[right] = buf[left]
+	end := blen - 1
+	for i := 0; i < blen; i++ {
+		res[i] = buf[end-i]
 	}
 	return res
 }
```
### Benchmarks before
```shell
$ go test -run=^$ -bench=. -count=10
goos: darwin
goarch: amd64
pkg: github.com/tendermint/go-crypto/hd
BenchmarkReverseBytes-4   	 5000000	       245 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       247 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       243 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       259 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       247 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       244 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       256 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       249 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       257 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       249 ns/op	     114 B/op	       4 allocs/op
PASS
ok  	github.com/tendermint/go-crypto/hd	15.166s
```
### Benchmarks after the full walk
```shell
$ go test -run=^$ -bench=. -count=10
goos: darwin
goarch: amd64
pkg: github.com/tendermint/go-crypto/hd
BenchmarkReverseBytes-4   	 5000000	       256 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       260 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       254 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       258 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       261 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       263 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       265 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       272 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       271 ns/op	     114 B/op	       4 allocs/op
BenchmarkReverseBytes-4   	 5000000	       260 ns/op	     114 B/op	       4 allocs/op
PASS
ok  	github.com/tendermint/go-crypto/hd	15.857s
```

## Comparisons
```shell
$ benchstat old.txt new.txt
name            old time/op    new time/op    delta
ReverseBytes-4     250ns ± 4%     262ns ± 4%  +4.97%        (p=0.000 n=10+10)

name            old alloc/op   new alloc/op   delta
ReverseBytes-4      114B ± 0%      114B ± 0%    ~     (all samples are equal)

name            old allocs/op  new allocs/op  delta
ReverseBytes-4      4.00 ± 0%      4.00 ± 0%    ~     (all samples are equal)
```